client/src/utils/ajax.ts

```diff
@@ -0,0 +1,223 @@
+import envData from '../../../config/env.json';
+import Tokens from 'csrf';
+import cookies from 'browser-cookies';
+
+const { apiLocation } = envData as { apiLocation: string };
+
+const base = apiLocation;
+const tokens = new Tokens();
+
+const defaultOptions: RequestInit = {
+  credentials: 'include'
+};
+
+// _csrf is passed to the client as a cookie. Tokens are sent back to the server
+// via headers:
+function getCSRFToken() {
+  const _csrf = typeof window !== 'undefined' && cookies.get('_csrf');
+  if (!_csrf) {
+    return '';
+  } else {
+    return tokens.create(_csrf);
+  }
+}
+
+async function get<T>(path: string): Promise<T> {
+  const res = await fetch(`${base}${path}`, defaultOptions);
+  return (await res.json()) as Promise<T>;
+}
```

Comment:
If this is returning a promise, we should return `res.json()`

Reply:
Good catch, I've reverted them to `.then()`. The `Promise<T>` is still needed since `.json()` returns `Promise<any>`.